### ISSUE TITLE
Import of organisations not working

### ISSUE THREAD

**Issue by maartenbeeckmans** [Lerentis/terraform-provider-gitea#1]
# System info

Terraform version v1.2.6
Gitea provider version v0.7.1
Gitea version 1.17.0 on Debian 11

# Problem

The Terraform provider crashes when trying to import an organization (could be other resources as well, not tested).

# Steps taken

## Create organization with Terraform

```tf
# Provider details omitted
resource "gitea_org" "terraform" {
  name       = "terraform"
  full_name  = "Terraform"
  visibility = "limited"
}
```

Terraform plan an apply works

## Delete state

Removed the local state file

The state had the following configuration

```json
{
  "version": 4,
  "terraform_version": "1.2.6",
  "serial": 3,
  "lineage": "91c3be8f-2dfe-8073-77d5-8c6ee56ac512",
  "outputs": {},
  "resources": [
    {
      "mode": "managed",
      "type": "gitea_org",
      "name": "terraform",
      "provider": "provider[\"registry.terraform.io/lerentis/gitea\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "avatar_url": "https://gitea.example.com/avatars/1b1ed905d54c18e3dd8828986c14be17",
            "description": "",
            "full_name": "Terraform",
            "id": "11",
            "location": "",
            "name": "terraform",
            "repo_admin_change_team_access": true,
            "visibility": "limited",
            "website": ""
          },
          "sensitive_attributes": [],
          "private": "bnVsbA=="
        }
      ]
    }
  ]
}
```

## Try importing that organization in Terraform state. The same resource is still present in the Terraform code as it should be.

Importing with id (see old state):

```sh
# terraform import gitea_org.terraform 11
gitea_org.terraform: Importing from ID "11"...
gitea_org.terraform: Import prepared!
  Prepared gitea_org for import
gitea_org.terraform: Refreshing state... [id=11]
╷
│ Error: Request cancelled
│
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-gitea_v0.7.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8dc652]

goroutine 85 [running]:
git.uploadfilter24.eu/terraform-provider-gitea/gitea.resourceOrgRead(0x0?, {0xc4c560?, 0xc0004c8a50})
	git.uploadfilter24.eu/terraform-provider-gitea/gitea/resource_gitea_organisation.go:29 +0x72
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xd756a0?, {0xd756a0?, 0xc0007844b0?}, 0xd?, {0xc4c560?, 0xc0004c8a50?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.20.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00045c460, {0xd756a0, 0xc0007844b0}, 0xc0003a1040, {0xc4c560, 0xc0004c8a50})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.20.0/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00011d1a0, {0xd756a0?, 0xc000784390?}, 0xc00011ec80)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.20.0/helper/schema/grpc_provider.go:613 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00017c320, {0xd756a0?, 0xc0003e7b90?}, 0xc000114a20)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/tf5server/server.go:747 +0x41e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xc20c00?, 0xc00017c320}, {0xd756a0, 0xc0003e7b90}, 0xc0003c6620, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.12.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000478000, {0xd78128, 0xc0003029c0}, 0xc0003bca20, 0xc00047a600, 0x11fa7d0, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb0b
google.golang.org/grpc.(*Server).handleStream(0xc000478000, {0xd78128, 0xc0003029c0}, 0xc0003bca20, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

Error: The terraform-provider-gitea_v0.7.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

> Using the name terraform instead of id 11 gives the same error

## Resources

Docs Terraform import: https://www.terraform.io/cli/commands/import

**Comment by Lerentis** [comment]
Thanks for the report. I am currently a but short on time but I will have a closer look over the weekend

**Comment by Lerentis** [comment]
fixed with [v0.7.2](https://github.com/Lerentis/terraform-provider-gitea/releases/tag/v0.7.2)
 
terraform import only knows the ID of the resource and i was referencing organisations by name as it was done the way in the gitea go sdk. Added a wrapper to search by ID and it should work now with `terraform import gitea_org.terraform 11`.

Thanks again for the report and feel free to reopen if i missed something :smiley:

**Comment by maartenbeeckmans** [comment]
Tested it and it works now.

Thanks for fixing it on such a short notice!
